From b96a021a828a576d1543609a47143468b6a67b04 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:21 -0400
Subject: [fs] ext4: renumber EXT4_EX_* flags to avoid flag aliasing problems

Message-id: <1414600485-1929-32-git-send-email-lczerner@redhat.com>
Patchwork-id: 98568
O-Subject: [RHEL 7.1 PATCH v2 31/55] ext4: renumber EXT4_EX_* flags to avoid flag aliasing problems
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit
d26e2c4d72c2f2a38246f618480864fe3224929c

 Suggested-by: Andreas Dilger <adilger@dilger.ca>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/ext4.h b/fs/ext4/ext4.h
index ffcde03..f6f90cb 100644
--- a/fs/ext4/ext4.h
+++ b/fs/ext4/ext4.h
@@ -572,7 +572,6 @@ enum {
 #define EXT4_GET_BLOCKS_NO_PUT_HOLE  0x0200
  /* Convert written extents to unwritten */
 #define EXT4_GET_BLOCKS_CONVERT_UNWRITTEN 0x0400
-/* DO NOT ASSIGN ADDITIONAL FLAG VALUES WITHOUT ADJUSTING THE FLAGS BELOW */
 
 /*
  * The bit position of these flags must not overlap with any of the
@@ -583,8 +582,8 @@ enum {
  * caching the extents when reading from the extent tree while a
  * truncate or punch hole operation is in progress.
  */
-#define EXT4_EX_NOCACHE    0x0800
-#define EXT4_EX_FORCE_CACHE   0x1000
+#define EXT4_EX_NOCACHE    0x40000000
+#define EXT4_EX_FORCE_CACHE   0x20000000
 
 /*
  * Flags used by ext4_free_blocks
-- 
1.7.1